lib/entities/ens_record.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:cake_wallet/ethereum/ethereum.dart';
+import 'package:cw_core/wallet_base.dart';
+import 'package:cw_core/wallet_type.dart';
+import 'package:ens_dart/ens_dart.dart';
+import 'package:http/http.dart';
+import 'package:web3dart/web3dart.dart';
+
+class EnsRecord {
+  static Future<String> fetchEnsAddress(String name, {WalletBase? wallet}) async {
```

Comment:
**Enhancement:**
`wallet` can't be null, you already have the null check `!` added before using it in `parse_address_from_domain` so you can remove the nullability if condition and so

Reply:
wallet should be nullable, having the ! there was a mistake
the intention being that this function be callable anywhere without the need for an existing wallet just to resolve an ens address